src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,131 @@
-export const PeopleTable = () => {
+import { Link, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { useMemo } from 'react';
+import { PersonLink } from './Links';
+import { Person } from '../types';
+import { Params, SearchWith } from '../utils/SearchWith';
+import { TypeSort } from '../types/TypeSort';
+
+type Props = {
+  people: Person[],
+  error: boolean,
+  searchParams: URLSearchParams,
+  query: string,
+  sexFilter: string,
+  centuries: string[],
+};
+
+function getFiltered(
+  people: Person[],
+  {
+    query,
+    sexFilter,
+    centuries,
+    sort,
+    order,
+  }: Params,
+) {
+  let filteredPeople = [...people];
+
+  if (sexFilter) {
+    filteredPeople = filteredPeople.filter(person => {
+      return person.sex === sexFilter;
+    });
+  }
+
+  if (query && typeof query === 'string') {
+    const correctQuery = query.trim().toLowerCase();
+
+    filteredPeople = filteredPeople.filter(person => {
+      return (
+        person.name.toLowerCase().includes(correctQuery)
+        || person.motherName?.toLowerCase().includes(correctQuery)
+        || person.fatherName?.toLowerCase().includes(correctQuery)
+      );
+    });
+  }
+
+  if (centuries
+    && Array.isArray(centuries)
+    && centuries.length !== 0) {
+    const correctCentury = centuries.map(century => +century);
+
+    filteredPeople = filteredPeople.filter(person => {
+      const bornCentury = Math.floor(person.born / 100) + 1;
+
+      return correctCentury.includes(bornCentury);
+    });
+  }
+
+  if (sort) {
+    filteredPeople = filteredPeople.sort((person1, person2) => {
+      switch (sort) {
+        case TypeSort.NAME:
+        case TypeSort.SEX:
+          return person1[sort].localeCompare(person2[sort]);
+
+        case TypeSort.BORN:
+        case TypeSort.DIED:
+          return person1[sort] - person2[sort];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
+    filteredPeople = filteredPeople.reverse();
+  }
+
+  return filteredPeople;
+}
+
+export const PeopleTable: React.FC<Props> = ({
+  people,
+  error,
+  searchParams,
+  query,
+  sexFilter,
+  centuries,
+}) => {
+  const { slug } = useParams();
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const visiblePeople = useMemo(() => {
+    return getFiltered(people, {
+      query,
+      sexFilter,
+      centuries,
+      sort,
+      order,
+    });
+  }, [searchParams]);
```

Comment:
add missing dependency
```suggestion
  }, [people, searchParams]);
```